core/src/main/java/com/linecorp/armeria/client/retry/RetryingClient.java

```diff
@@ -65,4 +84,66 @@ protected Backoff newBackoff() {
     protected RetryStrategy<I, O> retryStrategy() {
         return retryStrategy;
     }
+
+    /**
+     * Returns the number of requests sent and increase its count by one.
+     * This method should be invoked when the first request is made.
+     */
+    protected final int increaseNumRequestsSent(ClientRequestContext ctx) {
```

Comment:
After reading the code a little bit more, I realized that this is basically used to initialize the state of a request, and it returns a generated request ID.

- Could we rename this method again so it represents what it does better?
- You don't need to maintain a map, because you can attach an arbitrary object to a `RequestContext`. Please refer to `RequestContext.attr(...)` to simplify this even further.